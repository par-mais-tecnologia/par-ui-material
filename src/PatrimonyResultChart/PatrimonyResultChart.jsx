import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'

import { Grid, Typography } from '../index'
import { getCurrencyFormat, formatPercent } from '../Core/masks'

const styles = (theme) => ({
  patrimony: theme.patrimonyResultChart ? theme.patrimonyResultChart : {}
})

function PatrimonyResultChart (props) {
  const createObjects = (investments, realState, movableAssets, props) => {
    const total = investments + realState + movableAssets

    const percentinv = Number(((investments / total) * 100).toFixed(1)) || 0
    const percentreal = Number(((realState / total) * 100).toFixed(1)) || 0
    const percentmov = Number(((movableAssets / total) * 100).toFixed(1)) || 0

    const inv = {
      amount: getCurrencyFormat(investments, 'R$', 0, ' '),
      desc: 'Investimentos',
      percentage: `${formatPercent(percentinv)}`,
      backgroundColor: props.investmentColor,
      class: 'investment'
    }
    const rs = {
      amount: getCurrencyFormat(realState, 'R$', 0, ' '),
      desc: 'ImÃ³veis',
      percentage: `${formatPercent(percentreal)}`,
      backgroundColor: props.realStateColor,
      class: 'realstate'
    }
    const ma = {
      amount: getCurrencyFormat(movableAssets, 'R$', 0, ' '),
      desc: 'Outros bens',
      percentage: `${formatPercent(percentmov)}`,
      backgroundColor: props.movableAssetsColor,
      class: 'movable'
    }

    return [inv, rs, ma]
  }

  const investmentChart = (classes, property, key) => {
    return (
      <Grid key={key} className={`flex flex-row mb4 ${classes.patrimony}`}>
        <Grid className='flex flex-column' style={{ minWidth: 150 }}>
          <Typography
            variant='body1'
            className={`display4 onHover ${property.class}`}>
            {property.amount}
          </Typography>
          <Typography
            variant='body1'
            className={`onHover ${property.class}`}>
            {property.desc}
          </Typography>
          <Typography
            variant='body1'
            className={`percentage onHover ${property.class}`}>
            {String(property.percentage).replace(/\./, ',')}
          </Typography>
        </Grid>
        <Grid className='flex w-100' style={{ paddingLeft: 8 }}>
          <Grid
            className='bar'
            style={{
              width: property.percentage,
              backgroundColor: property.backgroundColor
            }}
          />
        </Grid>
      </Grid>
    )
  }

  const { classes, investments, realState, movableAssets } = props
  const patrimonies = createObjects(investments, realState, movableAssets, props)

  return (
    <div className='flex flex-column'>
      {patrimonies.map(inv => investmentChart(classes, inv, Math.floor(Math.random() * 6530)))}
    </div>
  )
}

PatrimonyResultChart.propTypes = {
  investments: PropTypes.number.isRequired,
  realState: PropTypes.number.isRequired,
  movableAssets: PropTypes.number.isRequired,
  investmentColor: PropTypes.string.isRequired,
  realStateColor: PropTypes.string.isRequired,
  movableAssetsColor: PropTypes.string.isRequired
}

export default withStyles(styles)(PatrimonyResultChart)
