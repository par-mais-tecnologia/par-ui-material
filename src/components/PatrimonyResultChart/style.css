@keyframes createBox {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(1.04);
  }
}