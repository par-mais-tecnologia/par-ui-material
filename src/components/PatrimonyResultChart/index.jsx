import React from 'react'
import Grid from '@material-ui/core/Grid'

import { Typography, withStyles } from '@material-ui/core'
import './style.css'

const styles = {
  patrimony: {
    paddingBottom: '16px',
    '&$patrimony:hover .onHover.investment': {
      color: '#632B7D'
    },
    '&$patrimony:hover .onHover.realstate': {
      color: '#5EB8C0'
    },
    '&$patrimony:hover .onHover.movable': {
      color: '#347A7C'
    },
    '&$patrimony:hover .bar': {
      animation: 'createBox 0.2s forwards'
    }
  }
}

function PatrimonyResultChart (props) {
  const investmentChart = (classes, property) => {
    return (
      <Grid container direction='row' className={classes.patrimony}>
        <Grid container direction='column' xs={6} alignItems='flex-end'>
          <Typography
            variant='display1'
            className={`onHover ${property.class}`}
          >
            {property.amount}
          </Typography>
          <Typography
            variant='body1'
            className={`onHover ${property.class}`}
          >
            {property.desc}
          </Typography>
          <Typography
            variant='display2'
            className={`onHover ${property.class}`}
          >
            {property.percentage}
          </Typography>
        </Grid>
        <Grid container xs={6} style={{ paddingLeft: '8px' }}>
          <Grid
            item
            className='bar'
            style={{
              width: property.percentage,
              backgroundColor: property.backgroundColor
            }}
          />
        </Grid>
      </Grid>
    )
  }

  const createObjects = (investments, realState, movableAssets) => {
    const total = investments + realState + movableAssets
    const percentinv = Number(((investments / total) * 100).toFixed(1))
    const percentreal = Number(((realState / total) * 100).toFixed(1))
    const percentmov = Number(((movableAssets / total) * 100).toFixed(1))
    const inv = {
      amount: `R$ ${investments}`,
      desc: 'Investimentos',
      percentage: `${percentinv}%`,
      backgroundColor: '#632B7D',
      class: 'investment'
    }
    const rs = {
      amount: `R$ ${realState}`,
      desc: 'Bens imóveis',
      percentage: `${percentreal}%`,
      backgroundColor: '#5EB8C0',
      class: 'realstate'
    }
    const ma = {
      amount: `R$ ${movableAssets}`,
      desc: 'Bens móveis',
      percentage: `${percentmov}%`,
      backgroundColor: '#347A7C',
      class: 'movable'
    }

    return [inv, rs, ma]
  }

  const { classes, investments, realState, movableAssets } = props
  const patrimonies = createObjects(investments, realState, movableAssets)

  return (
    <Grid container direction='column'>
      {patrimonies.map(inv => investmentChart(classes, inv))}
    </Grid>
  )
}

export default withStyles(styles)(PatrimonyResultChart)
