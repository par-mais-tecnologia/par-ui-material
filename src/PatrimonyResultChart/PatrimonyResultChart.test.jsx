import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import PatrimonyResultChart from './'

configure({ adapter: new Adapter() })

const foo = {
  totalInvestment: 0,
  immovableProperty: 0,
  movableProperty: 0
}

export const testShallow = describe('PatrimonyResultChart shallow', () => {
  const getShallow = () => (
    <PatrimonyResultChart
      investments={foo.totalInvestment}
      investmentColor={'#632B7D'}
      realState={foo.immovableProperty}
      realStateColor={'#5EB8C0'}
      movableAssets={foo.movableProperty}
      movableAssetsColor={'#347A7C'}
    />
  )

  it(`shallow PatrimonyResultChart`, () => {
    const wrapper = shallow(getShallow())
    console.log(wrapper.props())
    expect(wrapper).toBeTruthy()
  })

  it('shallow PatrimonyResultChart match Snapshot', () => {
    const wrapper = shallow(getShallow())
    expect(wrapper).toMatchSnapshot()
  })
})
