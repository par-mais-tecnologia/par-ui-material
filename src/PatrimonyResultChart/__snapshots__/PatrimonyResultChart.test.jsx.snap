// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatrimonyResultChart shallow shallow PatrimonyResultChart match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PatrimonyResultChart)
    investmentColor="#632B7D"
    investments={0}
    movableAssets={0}
    movableAssetsColor="#347A7C"
    realState={0}
    realStateColor="#5EB8C0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "patrimony": "PatrimonyResultChart-patrimony-1",
      },
      "investmentColor": "#632B7D",
      "investments": 0,
      "movableAssets": 0,
      "movableAssetsColor": "#347A7C",
      "realState": 0,
      "realStateColor": "#5EB8C0",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "patrimony": "PatrimonyResultChart-patrimony-1",
        },
        "investmentColor": "#632B7D",
        "investments": 0,
        "movableAssets": 0,
        "movableAssetsColor": "#347A7C",
        "realState": 0,
        "realStateColor": "#5EB8C0",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
